sds_data_manager/lambda_code/SDSCode/path_helper.py

```diff
@@ -92,39 +93,70 @@ def check_date_input(self, input_date: str) -> bool:
         bool
             Whether date input is valid or not
         """
-        # Check if the pattern matches 8 digits (YYYYMMDD)
-        if not re.match(r"^\d{8}$", input_date):
-            return False
 
         # Validate if it's a real date
         try:
+            # This checks if date is in YYYYMMDD format.
+            # Sometimes, date is correct but not in the format we want
+            if not re.match(r"^\d{8}$", input_date):
```

Comment:
This can just be removed, as the `strptime` will also fail if the input date isn't 8 digits exactly.

Reply:
I did remove it originally based on feedback from you and Greg. But when I input correct date such as `2023105`, `strptime` accepted because it took that as valid date.

Reply:
Odd! Well, good job verifying 😆